keycloak-claims-mapper/src/test/java/ee/eki/tolkevarav/sso/keycloakserviceprovider/tokenclaimsmapper/LocalKeycloakContainer.java

```diff
@@ -0,0 +1,42 @@
+package ee.eki.tolkevarav.sso.keycloakserviceprovider.tokenclaimsmapper;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.utility.MountableFile;
+
+import java.nio.file.Path;
+
+import static java.time.Duration.ofSeconds;
+
+public class LocalKeycloakContainer extends GenericContainer<LocalKeycloakContainer> {
```

Comment:
For unit tests, yes, it would have been possible – these tests would have been quite trivial. It’s more important here to test that the mapper _actually works_ in the context of authorization code flow. For this, spinning up our Docker image seemed like the simples solution. 

Perhaps I could’ve run some higher-level Keycloak classes with the mapper in the same classpath, but it is my opinion that this would have been more complex and less valuable for what it actually validates.

Reply:
> It’s more important here to test that the mapper actually works in the context of authorization code flow
I think its not important. Mappers only job is to add additional claims to access token, which means that the login already been successful, since the the plain accessToken is a requirement and from mappers perspective it is not important how the login procedure was made. In that sense simple direct grant would have been fine (request against `../protocol/openid-connect/token"` with grant_type `password`)

Reply:
👍